# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/locks/LockManager.cpp
# Commit: 9e364e2eda05
# Full Hash: 9e364e2eda05d2b22080a79ce291cd954946bfc3
# Author: Kagami Sascha Rosylight <krosylight@mozilla.com>
# Date: 2023-02-24 16:04:01
# Regressor Bug: 1818468
# File Overlap Count: 1
# Description:
#   Bug 1818468 - Throw `signal.reason` when already aborted in `navigator.locks.request` r=evilpie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D170742
# ==============================================================================

diff -r 398115c86eed -r 9e364e2eda05 dom/locks/LockManager.cpp
--- a/dom/locks/LockManager.cpp	Fri Feb 24 12:11:28 2023 +0000
+++ b/dom/locks/LockManager.cpp	Fri Feb 24 13:06:43 2023 +0000
@@ -5,6 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "mozilla/dom/LockManager.h"
+#include "mozilla/dom/AutoEntryScript.h"
 #include "mozilla/dom/WorkerCommon.h"
 #include "mozilla/dom/locks/LockManagerChild.h"
 #include "mozilla/dom/locks/LockRequestChild.h"
@@ -96,7 +97,13 @@
       return false;
     }
     if (options.mSignal.Value().Aborted()) {
-      aRv.ThrowAbortError("The lock request is aborted");
+      AutoEntryScript aes(options.mSignal.Value().GetParentObject(),
+                          "LockManager::Request");
+      JSContext* cx = aes.cx();
+      JS::Rooted<JS::Value> reason(cx);
+      options.mSignal.Value().GetReason(cx, &reason);
+      aRv.MightThrowJSException();
+      aRv.ThrowJSException(cx, reason);
       return false;
     }
   }